packages/common/src/api/user.ts

```diff
@@ -21,12 +22,30 @@ type GetUSDCTransactionListArgs = {
   method?: full.GetUSDCTransactionsMethodEnum
 }
 
-const parseTransaction = (
+/**
+ * Parser to reformat transactions as they come back from the API.
+ * @param transaction the transaction to parse
+ * @param rootSolanaAccount? Optionally a root solana account can be passed
+ *  to reformat the metadata field to include specific contextual information.
+ *  In the case of withdrawals, this is useful in recognizing a "self-send",
+ *  which is a cash transfer out.
+ */
+const parseTransaction = ({
+  transaction,
+  rootSolanaAccount
+}: {
   transaction: full.TransactionDetails
-): USDCTransactionDetails => {
-  const { change, balance, transactionType, method, ...rest } = transaction
+  rootSolanaAccount?: string
+}): USDCTransactionDetails => {
+  const { change, balance, transactionType, method, metadata, ...rest } =
+    transaction
   return {
     ...rest,
+    metadata: !rootSolanaAccount
+      ? metadata
+      : rootSolanaAccount === metadata.toString()
+      ? `Cash (${metadata})`
```

Comment:
nit: just making sure design/product is aligned on "Cash (wallet)" as opposed to "Bank"/"Coinflow"/"USD" etc